driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -421,14 +421,12 @@ private Connection findLeastBusyForShard(int shardId) {
       if (host.convictionPolicy.canReconnectNow()) {
         if (connectionsPerShard == 0) {
```

Comment:
It seems that this case when connectionsPerShard == 0 could be removed at all. I can't think of a case when it's true. @avelanarius @Lorak-mmk please verify.

Reply:
On a second thought it might be guarding against a configuration where core connections == 0 so it's better to keep it.